src/FileMigrationHelper.php

```diff
@@ -34,17 +36,32 @@ class FileMigrationHelper
      */
     private static $delete_invalid_files = true;
 
+    /**
+     * Specifies the interval for every Nth file looped at which output will be logged.
+     *
+     * @config
+     * @var int
+     */
+    private static $log_interval = 100;
+
+    /** @var LoggerInterface|null */
+    private $logger;
+
     /**
      * Perform migration
      *
      * @param string $base Absolute base path (parent of assets folder). Will default to PUBLIC_PATH
      * @return int Number of files successfully migrated
+     * @throws \Exception
      */
     public function run($base = null)
     {
+        $this->logger = Injector::inst()->get(LoggerInterface::class);
```

Comment:
Sorry to be pain, but for consistency properties should be initialised in the constructor.

Reply:
Jumping in as devil's advocate - yes, agreed, but SilverStripe's dependency injection (which we should try and use as much as possible) is lazily injected, not at construction. We should be careful not to promote two different design patterns in core

Reply:
More specific example of this would be that there's a setter for the logger and you define `private static $dependencies = ['Logger' => LoggerInterface::class]`

Reply:
As long as we keep the thing in a property it should be initialised in constructor (or through `private static $dependencies`). Otherwise, we end up with inconsistent state of the class, depending on the order of its method invocations.

> there's a setter for the logger and you define private static $dependencies

I guess that's good to have. However, depending on whether we want this class to be only available as DI service or constructable entity as well.

Reply:
Yeah that’s what I mean. While I’m not a fan of the DI we promote in framework, it is our recommended solution so I think we should continue to promote it until it gets replaced with something constructor based

Reply:
yeah. I'm not saying we should stop using Injector :) I'm rather saying that initialisation should be done in constructor (or by DI through a setter), not in the `run` method.

Reply:
I was having issues with $dependencies earlier. Seems to work now though, so I've made the necessary changes. :)